encoder.go

```diff
@@ -0,0 +1,13 @@
+package jwt
+
+// Base64Encoder is an interface that allows to implement custom Base64 encoding/decoding algorithms.
+type Base64Encoder interface {
```

Comment:
I wonder if it makes sense to split it into an Encoder and Decoder interface? And then name the options WithBase64Encoder and WithBase64Decoder for option and parser?

Reply:
It seems that at least https://github.com/bytedance/sonic#use-numberuse-int64 seems compatible to `encoding/json` so we could even support `UseNumber` with an interface like this:

```
type JSONDecoder interface {
  UseNumber()
  Decode(any) error
}
```

Reply:
Hmm it seems that using the decoder instead of `json.Unmarshal` is really slower (and takes slightly more allocations), so that's not a good idea. in this case maybe it would be worth it only to use NewDecoder if we really need it (e.g., if UseNumber is enabled) and only with the standard library.